core/sr-arithmetic/fuzzer/src/biguint.rs

```diff
@@ -0,0 +1,208 @@
+// Copyright 2019 Parity Technologies (UK) Ltd.
+// This file is part of Substrate.
+
+// Substrate is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Substrate is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
+
+//! # Running
+//! Running this fuzzer can be done with `cargo hfuzz run biguint`. `honggfuzz` CLI options can
+//! be used by setting `HFUZZ_RUN_ARGS`, such as `-n 4` to use 4 threads.
+//!
+//! # Debugging a panic
+//! Once a panic is found, it can be debugged with
+//! `cargo hfuzz run-debug biguint hfuzz_workspace/biguint/*.fuzz`.
+//!
+//! # More infomation
+//! More information about `honggfuzz` can be found
+//! [here](https://docs.rs/honggfuzz/0.5.45/honggfuzz/).
+
+use honggfuzz::fuzz;
+use sr_arithmetic::biguint::{BigUint, Single};
+use std::convert::TryFrom;
+use num_traits::ops::checked::CheckedDiv;
+
+type S = u128;
+
+fn main() {
+	loop {
+		fuzz!(|data: (Vec<Single>, Vec<Single>, bool)| {
+			let (mut digits_u, mut digits_v, rem) = data;
```

Comment:
I would immediately convert these two to 

```rust 
let u = BigUint::from_limbs(&digits_u);
let v = BigUint::from_limbs(&digits_v);
let ue = S::try_from(u.clone()).unwrap();
let ve = S::try_from(v.clone()).unwrap();
``` 

and clone for the rest of the function. I had to create them separately in `run_with_data_set` because they were separate functions at the time. You can for sure reduce some redundant code here

Reply:
🏴